apps/web/src/state/user/hooks/useUserAddedTokens.ts

```diff
@@ -16,3 +16,18 @@ export default function useUserAddedTokens(): Token[] {
   const { chainId } = useActiveChainId()
   return useSelector(useMemo(() => userAddedTokenSelector(chainId), [chainId]))
 }
+
+export const userAddedTokenSelectorByChainIds = (chainIds: number[]) =>
+  createSelector(selectUserTokens, (serializedTokensMap) =>
+    chainIds.reduce<{ [address: string]: Token[] }>((tokenMap, chainId) => {
+      /* eslint-disable no-param-reassign */
+      tokenMap[chainId] = Object.values(serializedTokensMap?.[chainId] ?? {}).map(deserializeToken)
+      return tokenMap
```

Comment:
```suggestion
      return {
        ...tokenMap,
        [chainId]: Object.values(serializedTokensMap?.[chainId] ?? {}).map(deserializeToken)
      }
```

Reply:
there will be performance issue because each iterate will make a new copy